### PR TITLE
Update DG user stories, use cases and manual testing sections and some test changes

### PR DESCRIPTION
**Pull request by TanLeYang**
- Updated the above mentioned sections of the DG 
- Some refinements to the tests for `Name`, `Genre` and `Tag`
- Add tests for `ViewTypeManager`

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -765,29 +815,135 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test case: `add n/Test book g/T@st genre tp/1000 b/50`<br>
+      Expected: No book is added. Error details shown in the status message.
+   
+   1. Other incorrect add commands to try: `add n/Test g/Test tp/X b/Y` where Y is larger than X <br>
+      Expected: Similar to previous
+
+2. Adding a book while only some books are being shown.
+
+   1. Test case: `add n/Test book g/TEst genre tp/1000 b/50`<br>
```
Comment (pennhanlee) [inline docs/DeveloperGuide.md]: There is a typo with Test genre

**Review by pennhanlee — COMMENTED** [review]
LGTM besides a small typo in DG

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (TanLeYang) [reply docs/DeveloperGuide.md]: Oh right nice catch!

**Review by pangpuncake — APPROVED** [review]
LGTM!

**Review by angrybunny123 — APPROVED** [review]
LGTM 👍

**Review by mgiang2015 — COMMENTED** [review]
LGTM!

**Pull request merged**